pkg/collector/corechecks/net/network/network.go

```diff
@@ -308,13 +709,18 @@ func (c *NetworkCheck) Configure(senderManager sender.SenderManager, _ uint64, r
 }
 
 // Factory creates a new check factory
-func Factory() option.Option[func() check.Check] {
+func Factory() option.Option[func(config.Component) check.Check] {
```

Comment:
```suggestion
func Factory(config config.Component) option.Option[func() check.Check] {
        procfsPath := "/proc"
	if cfg.IsConfigured("procfs_path") {
		procfsPath = strings.TrimRight(cfg.GetString("procfs_path"), "/")
	}

	return option.New(func() check.Check {
		return &NetworkCheck{
		        net: defaultNetworkStats{procPath: procfsPath},
			CheckBase:     core.NewCheckBase(CheckName),
			coreconfig: config,
		}
	})
}
```

Note: that we need to add the `coreconfig` filed to the NetworkCheck struct

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/33190/commits/6cb3260b4969492040c7754bd997f740c8e99247 , I was running into the issue because of the stub.go function signature